akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ClusterSharding.scala

```diff
@@ -309,6 +310,12 @@ class ClusterSharding(system: ExtendedActorSystem) extends Extension {
     } else {
       log.debug("Starting Shard Region Proxy [{}] (no actors will be hosted on this node)...", typeName)
 
+      if (settings.shouldHostCoordinator(cluster)) {
+        // when using another coordinator role than the shard role the coordinator may have to be started
+        val startCoordinatorMsg = StartCoordinatorIfNeeded(typeName, settings, allocationStrategy)
+        guardian ! startCoordinatorMsg
+      }
```

Comment:
That is right, if it's not a "shard role" it will start a proxy, but that may also be a "coordinator role". The StartProxy message doesn't contain all needed parameters and that is also used in from elsewhere. Therefore I added this as a separate message.